### PR TITLE
param differences docs

### PR DESCRIPTION
**Pull request by steinroe**
Added docs about differences in the params and default values for RDSE Encoder, TM and SP.

### CODE REVIEW

File: PARAM_DIFFERENCES.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+## Parameter Differences
+
```
Comment (breznak) [inline PARAM_DIFFERENCES.md:2]: please add a brief explanatory note: "This is a comparison of parameters for python versions of this repository htm.core (link) and Numenta's NuPIC (link). Along with the API_CHANGELOG (link) this serves as a guide when migrating your code or comparing the frameworks. "

File: PARAM_DIFFERENCES.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+### RDSE
+| HTMCore    |                                                                                                                                                                                                                                                                                            |              | Nupic CPP  |                                                                                                                                                                                                                                                                                                                 |         |
```
Comment (breznak) [inline PARAM_DIFFERENCES.md:4]: not Nupic CPP (that was our former name), ratner NuPIC, or Numenta's NuPIC

File: PARAM_DIFFERENCES.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+| size       | Member "size" is the total number of bits in the encoded output SDR\.                                                                                                                                                                                                                      | 0            | n          | Number of bits in the representation \(must be > w\)\. n must be large enough such that there is enough room to select new representations as the range grows\. With w=21 a value of n=400 is typical\. The class enforces n > 6\*w\.                                                                           | 400     |
```
Comment (breznak) [inline PARAM_DIFFERENCES.md:7]: nit: if you want, you could enrich the doc (c++ .hpp, py) of the arguments with "(say for size) This used to be `n` in old NuPIC."

File: PARAM_DIFFERENCES.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| Attribute                 | Description                                                                                                                                                                                                                                                                                                                                                                                                          | Default     | Attribute                 | Description | Default |
+| columnDimensions          | Dimensions of the column space                                                                                                                                                                                                                                                                                                                                                                                       | \-          | columnDimensions          |             | \-2048  |
```
Comment (breznak) [inline PARAM_DIFFERENCES.md:41]: typo -2048 :)

File: PARAM_DIFFERENCES.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+| initialPermanence         | Initial permanence of a new synapse\.                                                                                                                                                                                                                                                                                                                                                                                | 0\.21       | initialPermanence         |             | 0\.21   |
+| connectedPermanence       | If the permanence value for a synapse is greater than this value, it is said to be connected\.                                                                                                                                                                                                                                                                                                                       | 0\.5        | connectedPermanence       |             | 0\.5    |
```
Comment (breznak) [inline PARAM_DIFFERENCES.md:45]: nit: probobly we won't cover that here, but there's a diff between what's in the defaults param, and what's actually used as a default everywhere (tests, NAB,..). For connectedPermanence it's 0.2 for Numenta.

**Comment by breznak** [comment]
FYI @dkeeney , keep as is in a separate file, or add directly to the README as Appendix?

**Comment by dkeeney** [comment]
> keep as is in a separate file, or add directly to the README as Appendix?

Keeping it as a separate file would give it more exposure as long as we link to it from the README.

File: PARAM_DIFFERENCES.md (hunk 5 above)
Comment (steinroe) [reply PARAM_DIFFERENCES.md:45]: I have another column "Value in NAB" for both, HTMCore as well as Numenta's NuPiC, but I removed it. I can add it back into the table, but I would propose to wait until we have the param set for HTMCore optimised.

File: PARAM_DIFFERENCES.md (hunk 2 above)
Comment (steinroe) [reply PARAM_DIFFERENCES.md]: Done 👍

File: PARAM_DIFFERENCES.md (hunk 3 above)
Comment (steinroe) [reply PARAM_DIFFERENCES.md]: Done 👍

File: PARAM_DIFFERENCES.md (hunk 4 above)
Comment (steinroe) [reply PARAM_DIFFERENCES.md]: Thanks 👍

File: PARAM_DIFFERENCES.md (hunk 1 above)
Comment (steinroe) [reply PARAM_DIFFERENCES.md]: Done 👍

**Comment by steinroe** [comment]
> Great work, thank you! 👍
> A couple of changes, please, and a few for consideration only.
> 
> * [ ]  I'd like this doc to be more visible, please link through the main README. Add to the line:
> 
> > If you are porting your code to this codebase, please review the API Changelog.
> 
> "... please follow the API_DIFFERENCES (link) and consult the API_CHANGELOG (link)"
> 
> * [ ]  for consideration, if we want to shorten the list by enlisting only the differences (?) (in comment, meaning, default value)

I added it to the README. I thought about that too. The downside of listing only the differences would be that sometimes the doc strings for the params differ and the differences might be of interest for some too. Also, when we add another column "Values in NAB", almost every attribute is is going to be different in that respect.

File: PARAM_DIFFERENCES.md (hunk 5 above)
Comment (breznak) [reply PARAM_DIFFERENCES.md]: > I have another column "Value in NAB" for both 

yes, "NAB column" is a good idea! 
And yes, let's wait where we get there.

File: API_DIFFERENCES.md [hunk 6]
```diff
@@ -1,29 +1,31 @@
 ## Parameter Differences
+This is a comparison of parameters for python versions of this repository htm.core (link) and Numenta's NuPIC (link). Along with the API_CHANGELOG (link) this serves as a guide when migrating your code or comparing the frameworks.
```
[the remark this answers is not in the document]
Comment (steinroe) [reply API_DIFFERENCES.md]: sorry I missed these. Added them now.

**Pull request merged**